apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -1,42 +1,355 @@
+import { z } from 'zod';
+import slugify from 'slugify';
 import { useNavigate, useParams } from 'react-router-dom';
-import { useEffectOnce } from '../../hooks';
+import { Controller, useForm } from 'react-hook-form';
+import { Button, colors, Input, NameInput, Switch, Text } from '../../design-system';
+import { Check, Close, Copy, DisconnectGradient } from '../../design-system/icons';
 import { useProviders } from './useProviders';
 import { IIntegratedProvider } from './IntegrationsStoreModal';
+import { IntegrationInput } from './components/IntegrationInput';
+import { IntegrationChannel } from './components/IntegrationChannel';
+import { CHANNEL_TYPE_TO_STRING } from '../../utils/channels';
+import { IntegrationEnvironmentPill } from './components/IntegrationEnvironmentPill';
+import { useFetchEnvironments } from '../../hooks/useFetchEnvironments';
+import { ProviderImage } from './components/multi-provider/SelectProviderSidebar';
+import { When } from '../../components/utils/When';
+
+interface IProviderForm {
+  name: string;
+  credentials: Record<string, string>;
+  active: boolean;
+  identifier: string;
+}
+
+const schema = z
+  .object({
+    identifier: z
+      .string({
+        required_error: 'Required - Instance key',
+      })
+      .superRefine((data, ctx) => {
+        const regexp = new RegExp(/^[A-Za-z_-]+$/);
+        if (!regexp.test(data)) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.invalid_string,
+            validation: 'regex',
+            message: 'Instance key must only contains alphabetical, dash and underscore characters',
+          });
+        }
+      }),
+  })
+  .passthrough();
 
 export function UpdateProviderPage() {
+  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
   const [selectedProvider, setSelectedProvider] = useState<IIntegratedProvider | null>(null);
-  const { providers } = useProviders();
+  const { providers, refetch, isLoading } = useProviders();
   const { colorScheme } = useMantineColorScheme();
   const isDark = colorScheme === 'dark';
   const { integrationId } = useParams();
   const navigate = useNavigate();
 
-  useEffectOnce(() => {
+  const {
+    control,
+    handleSubmit,
+    reset,
+    watch,
+    setValue,
+    formState: { errors, isDirty },
+  } = useForm<IProviderForm>({
+    shouldUseNativeValidation: false,
+    resolver: zodResolver(schema),
+    defaultValues: {
+      name: '',
+      credentials: {},
+      active: false,
+      identifier: '',
+    },
+  });
+
+  const credentials = watch('credentials');
+  const haveAllCredentials = useMemo(() => {
+    if (selectedProvider === null) {
+      return false;
+    }
+    const missingCredentials = selectedProvider.credentials
+      .filter((credential) => credential.required)
+      .filter((credential) => {
+        const value = credentials[credential.key];
+
+        return value === undefined || value === null || value.length === 0;
```

Comment:
```suggestion
        return !value;
```
This would return `true` for these cases: undefined, null, 0 and '' (that I think is missing, just think someone that introduced and empty string key).